### PR TITLE
Move > and >_ from David A. Wheeler's mathbox to main; use discouraged

### PR DESCRIPTION
**Pull request by david-a-wheeler**
Per David A. Wheeler's email on Sun, 01 Jan 2017 14:42:45 -0500 (EST),
move greater-than ` > ` and greater-than-or-equal ` >_ ` into the
main body with all relevant uses marked as "(New usage is discouraged.)".

"Greater-than" and "greater-than-or-equal" are widely used.  Even in
set.mm the term "greater" is used 185 times when you ignore this my
greater-than section.  I'd like set.mm to include formal definitions
of all significant mathematical terms (since I view its ultimate
goal as "formally proving all mathematics using typical ZFC axiomatic
conventions").  In that light, omitting greater-than is a problem.

The old justification for *not* moving it up is reasonable enough:
Every use creates additional work, so the convention is to always use <
and <_. Obviously, omitting ">" and ">_" from the main body ensures that
"<" and "<_" are always used instead.  But these are now marked
with the newer marking "(New usage is discouraged.)".
With that marking, I believe that problem goes away. This
marking will discourage any use of > and >_ in new theorems. In addition,
even if that is ignored, the set.mm integration tests now *specifically*
check to make sure that all changes to the "discouraged" list are
noticed. That means that > and >_ can be moved into the main body
*without* any risk that they will be unintentionally used.

The examples were moved into the "examples" section.

### CODE REVIEW

**Comment by digama0** [comment]
I'm thinking of rejecting this. It doesn't seem to gain us much, particularly since we still don't want people to use it. Most of the existing (New usage is discouraged.) theorems are there because they _are_ needed for something, possibly in a way which can't easily be removed. How about a link in df-ltxr instead? By being in a mathbox, it already achieves the stated purpose, which is to have a definition of the concept in the file, which can be used as a target in hyperlinks whenever needed.

**Comment by david-a-wheeler** [comment]
One problem is that leaving it in a mathbox doesn't provide any logical structure.  Certainly ` > ` and ` >_ ` are not unique to me :-). I think they should be defined near ` < ` , because they are logically related.

As far as "gain us much" - I think it does.  If someone wants to know "what does X mean" for a common mathematical construct, he can find a formal definition of it, and in the main body.

I think many people will consider the mathbox sections as "less authoritative" - which makes sense, because a lot of work-in-progress material is there.  In this case, though, I don't think that ` < ` is ever going to have a different definition; that *is* its definition.

In short: It won't increase the size of set.mm, it creates an authoritative definition of common math symbols where they logically below, and the "discouraged" markings will inhibit their later use (which was the original concern).

**Comment by nmegill** [comment]
I have thought about this too over the last couple of days and have very 
mixed feelings about it, because it goes against our standard practice. 
 From a mathematical point of view, > is really not a new definition but 
just a typographical variant of < that swaps arguments.  Our general 
philosophy on definitions is to add them when they offer a mathematical 
benefit such as shorter proofs.  There are many notations that have 
variants, and we pick one and document the others in the description 
such as in df-r1.  There is no reason we can't do the same with >.

Almost all ordering signs, from membership e. to subset to dominance, 
are freely mixed with the reversed version in textbooks.  To be 
consistent we would have to add those as well.  I don't think that an 
encyclopedia of all possible symbol variants in set.mm is a useful goal. 
  Our practice has been to select a single symbol from the ones 
available for a concept and treat it as the canonical symbol for that 
concept.

Having the > symbol defined in the main set.mm, but at the same time 
discouraged from being used, seems to give a mixed message.  Even if we 
just defined > in English in the < description, with no formal 
definition at all, it is hard to see how anyone would not know what is 
meant.

Norm

**Comment by david-a-wheeler** [comment]
> " > is really not a new definition but just a typographical variant of < that swaps arguments." ... 

Usually "typographical variant" means varying the appearance of the same symbol (e.g., Arial vs. Times New Roman).  I think most people would consider ` > ` to be a different symbol, not just a "typographical variant".  You can't just substitute one symbol for the other one - you have to swap the arguments.

> I don't think that an encyclopedia of all possible symbol variants in set.mm is a useful goal.

I agree that "all possible" isn't worth it.  But I think that "all common symbols" should have a formal definition somewhere in the body of the document.  Clearly ` > ` is common.

> Our practice has been to select a single symbol from the ones available for a concept and treat it as the canonical symbol for that concept. Having the > symbol defined in the main set.mm, but at the same time discouraged from being used, seems to give a mixed message.

I don't think so.  We now strongly enforce the "discouraged" list, both with individual tools and in the set.mm check-in.  It'd be hard to think that its use it *not* discouraged.

> Even if we just defined > in English in the < description, with no formal definition at all, it is hard to see how anyone would not know what is meant.

Doing it that way means that the symbol does not show up in the symbol list we can't make checked proofs (e.g., that "A < B" is equivalent to "B > A").

**Comment by nmegill** [comment]
On 1/7/17 5:22 PM, David A. Wheeler wrote:
>     " > is really not a new definition but just a typographical variant
>     of < that swaps arguments." ...
>
> Usually "typographical variant" means varying the appearance of the same
> symbol (e.g., Arial vs. Times New Roman). I think most people would
> consider |>| to be a different symbol, not just a "typographical
> variant". You can't just substitute one symbol for the other one - you
> have to swap the arguments.

I guess I meant "graphical variant".  The point is that a different 
symbol with a different order of arguments doesn't create a new 
mathematical concept.

>     I don't think that an encyclopedia of all possible symbol variants
>     in set.mm is a useful goal.
>
> I agree that "all possible" isn't worth it. But I think that "all common
> symbols" should have a formal definition somewhere in the body of the
> document. Clearly |>| is common.

So where do we draw the line?  We also have 4 or 5 common symbols for 
division (/, ÷, :,  written as a fraction, long division symbol).  The 
last one has the arguments reversed.

>     Even if we just defined > in English in the < description, with no
>     formal definition at all, it is hard to see how anyone would not
>     know what is meant.
>
> Doing it that way means that the symbol does not show up in the symbol
> list we can't make checked proofs (e.g., that "A < B" is equivalent to
> "B > A").

Since it is in your mathbox, it will be in the symbol list.

I doubt many people look up concepts by symbol.  Most symbols are 
Unicode that are hard to type, and sometimes the chance of matching 
numerous Unicode variants may be poor.  I think most people use 
English-language searches, which will match both "less than" and 
"greater than".

Norm

**Comment by david-a-wheeler** [comment]
> So where do we draw the line?  We also have 4 or 5 common symbols for division (/, ÷, :,  written as a fraction, long division symbol).  The last one has the arguments reversed.

A useful starting rule might be: "when spoken, do you say it differently?"  As far as I know, the symbols you listed are all pronounced as "division".  In contrast, people really do say "greater-than" and "less-than" differently.

**Comment by jkingdon** [comment]
In textbooks (and even in some other theorem provers), > versus < doesn't really affect how you write or interact with proofs. In metamath, you would need to add extra proof steps to make use of >. Therefore, I think this is better handled by text informally mentioning the > notation rather than adding a formal definition for >.

Adding this and saying it is discouraged does seem like an uneasy compromise to me.

**Comment by david-a-wheeler** [comment]
the direction of the symbol makes a big difference in a proof, obviously x < 0 is not the same as x > 0. It's true that you can write x > 0 as 0 < x ... but > is an extremely commons symbol, one that is widely used in mathematics. It's weird to not to define that formally.

**Comment by nmegill** [comment]
While we have had this discussion before, sometimes there is a conflict 
between how a user may want to have the math displayed and the burden 
such that a redundant notation places on the database.  There are around 
1300 theorems involving "<", about half of which could be considered 
"library" theorems, so to have an equally convenient library for ">" we 
would have to add around 600 more.  That doesn't include having all 
possible combinations of "<" and ">" in a given theorem, which would 
grow that number to many thousands (a sizeable percentage of the current 
30K theorems in set.mm).  For this reason, ordinarily we don't define 
two ways of expressing the same thing if there are no savings in 
expression size (like there would be with say df-an).

OTOH it is nice to have the > symbol available for use in a theorem's 
description if it is felt useful for clarity or to match a textbook 
description.  The math-style comment markup currently has available only 
the symbols defined for the actual math (and we don't plan to change 
that), so having > defined can be convenient for comments.

As a compromise, we have agreed to let David define >, but with the 
comment in its description (actually a markup command) "(New usage is 
discouraged.)"  This is a stronger directive than the word "discouraged" 
suggests:  you are actually forbidden to use it in the 2 main proof 
assistants unless you go out of your way to override it.  And if you do 
override it, there will be a failure in the Travis check until you 
regenerate a file called "discouraged", giving the github manager 
another opportunity to accept or reject it.

Norm

**Comment by nmegill** [comment]
I see now that the pull request, which has been open for some time, 
hasn't been approved (although the symbol is defined in DAW's mathbox). 
Anyway I would go along with the "(New usage is discouraged.)" 
compromise if others agree.  So I'll wait for Mario's agreement to make 
the final decision.

Norm

**Comment by david-a-wheeler** [comment]
@nmegill - I agree, that was my thinking as well. As long as it is discouraged, you really have to work to use it, which is the point.   I think we should encourage people to follow the conventions, but at least define this extremely common symbol. It is also useful for comments... if greater than is defined, then you can get nice typography for it.

**Comment by digama0** [comment]
I vote for leaving it where it is. So far I haven't seen any voices other than @david-a-wheeler in favor of this proposal, and all the benefits of using and referring to it in comments are already gained by having it in a mathbox. As a theorem development, it is a small thing that we have no interest in enlarging or becoming a dependency for something else, but it's just natural enough that people might use it accidentally if it's there. Best to just head off the whole issue by making it impossible to use (in theorems) in main.

**Comment by avekens** [comment]
According to #1389, there will be no chance that this pull request will be accepted in the near future. Is it possible to transform this pull request into an issue or mark it as "deferred"? As for the other pull requests #65 and #94, it seems to be strange that they are open for such a long time. Pull requests should be either accepted or rejected after a "short" time, let's say after 3 or 6 months latest.

**Comment by benjub** [comment]
I do not have strong opinions, but I think that indeed, > and $\geq$ are
pretty useless. Next step is defining $\ni$ ?

**Comment by david-a-wheeler** [comment]
We are currently having a discussion about the purpose of set.mm, it seems premature to close this without that discussion.

As I posted separately, I think it is important to Define formally all widely used mathematical symbols, in which case this is absolutely vital and should be included.

**Comment by jkingdon** [comment]
I guess I don't object to any of the proposed solutions (leave > in the mathbox where it is now, add an "Other Definitions" section after the mathboxes, or go ahead and merge this with the discouraged tags as the caveat that we don't want to prove every theorem for both < and >). But I do like the idea of closing out the pull request somehow for all the "hanging pull request" reasons.

**Comment by david-a-wheeler** [comment]
@jkingdon - fair enough. I am waiting for any stragglers in the conversation on the mailing list.I expect to close this pull request by this weekend.

**Comment by david-a-wheeler** [comment]
PR dropped per mailing list discussion.
